pkgs/development/compilers/swi-prolog/default.nix

```diff
@@ -15,20 +15,30 @@ stdenv.mkDerivation {
   };
 
   buildInputs = [ gmp readline openssl libjpeg unixODBC libXinerama
-    libXft libXpm libSM libXt zlib freetype pkgconfig fontconfig ];
+    libXft libXpm libSM libXt zlib freetype pkgconfig fontconfig ]
+  ++ stdenv.lib.optional stdenv.isDarwin makeWrapper;
 
   hardeningDisable = [ "format" ];
 
   configureFlags = "--with-world --enable-gmp --enable-shared";
 
   buildFlags = "world";
 
+  # still not fixed in upstream: abort trap 6 when called through
+  # symlink on macos, so wrap binary
+  postInstall = stdenv.lib.optionalString stdenv.isDarwin ''
+    local prog="$out/bin/swipl"
+    local hidden="$(dirname "$prog")/.$(basename "$prog")"-wrapped
+    mv $prog $hidden
```

Comment:
@copumpkin @LnL7 wrapProgram additionaly adds '-a "$0"' to exec and that breaks PAKCS build: https://github.com/NixOS/nixpkgs/blob/master/pkgs/build-support/setup-hooks/make-wrapper.sh